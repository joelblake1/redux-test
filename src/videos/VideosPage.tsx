import React from 'react';
import { connect } from 'react-redux';
import { GetVideosThunk } from '../redux/actions';
import { Video } from '../redux/models';
import { RootState } from '../redux/state';

export interface VideoProps{
  fetchVideos(): void;
  videos: Video[];
}

export const VideosComponent = (props: VideoProps): JSX.Element =>{
  React.useEffect(()=>{
    props.fetchVideos();
  });
  
  return (
    <>
      <form>
        <h2>Videos</h2>
      </form>
      <div>{props.videos.map(v =>
        <div>{v.name}</div>
        )}
      </div>    
    </>
    );
}

const mapState = (state: RootState) => ({
  videos: state.video.videos
})

const mapDispatch = (dispatch:any) => ({  
  fetchVideos() {
    dispatch(GetVideosThunk());
  }
})

export const ConnectedVideosComponent = connect(mapState, mapDispatch)(VideosComponent);
