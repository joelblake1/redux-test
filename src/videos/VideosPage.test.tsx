import { act } from "react-dom/test-utils";
import { VideoProps, VideosComponent } from "./VideosPage";

describe("VideosPage Should", ()=>{
  const videosProps: VideoProps = {videos:[{name: "first video", key: "asd"}], fetchVideos: jest.fn()}
  
  it("match snapshot", () =>{
    const videosPage = <VideosComponent {...videosProps}/>
    expect( videosPage).toMatchSnapshot()
  });
});
