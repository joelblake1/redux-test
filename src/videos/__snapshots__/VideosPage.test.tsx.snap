// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosPage Should match snapshot 1`] = `
<VideosComponent
  fetchVideos={[MockFunction]}
  videos={
    Array [
      Object {
        "key": "asd",
        "name": "first video",
      },
    ]
  }
/>
`;
